src/core/tests/FileSystemTestSuite.cpp

```diff
@@ -29,62 +33,155 @@ namespace {
                        return wPath == wFileName || wPath == wFileName2;
                      });
 }
+
+void createFile(const std::string& filePath, std::size_t sizeInBytes = 4096) {
+  std::ofstream outFile(filePath);
+
+  if(!outFile) {
+    throw std::runtime_error("Unable to open file: " + filePath);
+  }
+
+  // Write zeros to the file in chunks
+  constexpr std::size_t chunkSize = 4096;    // Write in 4 KB chunks
+  std::vector<char> buffer(chunkSize, 0);
+
+  std::size_t writtenBytes = 0;
+  while(writtenBytes < sizeInBytes) {
+    std::size_t bytesToWrite = std::min(chunkSize, sizeInBytes - writtenBytes);
+    outFile.write(buffer.data(), bytesToWrite);
+    writtenBytes += bytesToWrite;
+  }
+
+  outFile.close();
+  if(!outFile.good()) {
+    throw std::runtime_error("Error occurred while writing to file: " + filePath);
+  }
+}
 }    // namespace
 
 TEST(FileSystem, findCppFiles) {
+#ifndef _WIN32
```

Comment:
why?1

Reply:
I will try to modify test cases to be portable.